scripts/doc_checker.py

```diff
@@ -24,97 +40,242 @@
-    import_path : str
-        Python-like import path.
+    results : dict
+        Dictionary that numpydoc.validate.validate return.
+    modin_errors : list
+        List of tuples with Modin error code and its description.
+
+    Returns
+    -------
+    dict
+        Updated dict with Modin custom errors.
+    """
+    for error in modin_errors:
+        results["errors"].append(error)
+    return results
+
+
+def skip_check_if_noqa(doc: Docstring, err_code: str, noqa_checks: list) -> bool:
+    """
+    Skip the check that matches `err_code` if `err_code` found in noqa string.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Docstring
+        Docstring handler.
+    err_code : str
+        Error code found by numpydoc.
+    noqa_checks : list
+        Found noqa checks.
 
     Returns
     -------
     bool
         Return True if 'noqa' found.
     """
-    import inspect
-    from numpydoc.validate import Docstring
+    if noqa_checks == ["all"]:
+        return True
 
-    result = False
-    doc = Docstring(import_path)
-    source = doc.method_source
+    if err_code == "GL08":
+        name = doc.name.split(".")[-1]
+        magic = name.startswith("__") and name.endswith("__")
+        if inspect.isclass(doc.obj):
+            err_code = "D101"
+        elif inspect.ismethod(doc.obj):
+            err_code = "D102"
+        elif inspect.isfunction(doc.obj) and not magic:
+            err_code = "D103"
+        elif inspect.isfunction(doc.obj) and magic:
+            err_code = "D105"
 
-    noqa_str = None
-    # find last line of obj definition
-    for line in source.split("\n"):
-        if ")" in line and ":" in line.split(")", 1)[1]:
-            noqa_str = line
-            break
+    return err_code in noqa_checks
 
-    if noqa_str and "noqa" in noqa_str:
-        if (
-            "noqa:" not in noqa_str
-            or ("D102" in noqa_str and inspect.ismethod(doc.obj))
-            or ("D105" in noqa_str and inspect.isfunction(doc.obj))
-            or ("D103" in noqa_str and inspect.isfunction(doc.obj))
-            or ("D101" in noqa_str and inspect.isclass(doc.obj))
-        ):
-            result = True
 
-    return result
+def get_noqa_checks(doc: Docstring) -> list:
+    """
+    Get codes after `# noqa`.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Docstring
+        Docstring handler.
+
+    Returns
+    -------
+    list
+        List with codes.
+
+    Notes
+    -----
+    If noqa doesn't have any codes - returns ["all"].
+    """
+    source = doc.method_source
+    noqa_str = None
+    if not inspect.ismodule(doc.obj):
+        # find last line of obj definition
+        for line in source.split("\n"):
+            if ")" in line and ":" in line.split(")", 1)[1]:
+                noqa_str = line
+                break
+    else:
+        # pydocstyle only check first line; aling with it
+        noqa_str = source.split("\n", 1)[0]
+    if "noqa" not in noqa_str:
```

Comment:
I would advise using regexp-s here, as this one currently doesn't check for pound sign (`#`), so it would fail for cases like
```python
def foo(value="noqa: MD01"): # my super function
    return 10
```